P1684-mdarray/P1684.md

```diff
@@ -334,6 +342,397 @@ In R3 we decided to radically cut back on constructors and simply leave those ou
 
 Finally, `mdarray` does not have the analog of `mdspan`'s constructor that takes an `array<IndexType, N>` of dynamic extents.  This avoids any ambiguity or confusion with a constructor that takes a container instance, in the case where the `container` happens to be an `array<IndexType, N>`.
 
+### Alternative constructor design
+
+We can both significantly reduce the number of constructors *and*
+gain the ability to emplace the container.  This requires
+a bit of shuffling of the arguments, adding a tag and adding a parameter pack.
+
+* Whenever `extents_type` or `mapping_type` is provided, it must come first.
+* Whenever an `mdarray` is constructed with `val`, it must be preceded by a tag.
+* Add a parameter pack `Args...` that essentially emplace constructs the container.
+
+The original adaptors couldn't do this, because variadics didn't exist in C++98.
```

Comment:
Would you consider elaborating "the original adapters" here?

```suggestion
The original container adaptors in the C++98 Standard Library couldn't do this, because variadic template parameters didn't exist in C++98.
```

Reply:
I'll rephrase it as "the container adaptors introduced in C++98".  They aren't 100% the same because of move semantics (which leads to the current problem with `priority_queue` not guaranteed to maintain its invariants).